zap/src/main/java/org/zaproxy/zap/extension/alert/AlertViewPanel.java

```diff
@@ -333,6 +341,40 @@ public void keyPressed(java.awt.event.KeyEvent evt) {
                         }
                     });
 
+            AlertTagsTableModel alertTagsTableModel = new AlertTagsTableModel(editable);
+            alertTags = new ZapTable(alertTagsTableModel);
+            alertTags.setPreferredScrollableViewportSize(
+                    new Dimension(
+                            alertTags.getPreferredSize().width, alertTags.getRowHeight() * 5));
+            alertTags.setCellSelectionEnabled(true);
+            alertTags.setEditable(editable);
+            if (editable) {
+                alertTags.addMouseListener(
+                        new java.awt.event.MouseAdapter() {
+                            @Override
+                            public void mouseClicked(java.awt.event.MouseEvent evt) {
+                                int row = alertTags.rowAtPoint(evt.getPoint());
+                                int col = alertTags.columnAtPoint(evt.getPoint());
+                                if (col == 0 && row == alertTags.getRowCount() - 1) {
+                                    alertTagsTableModel.addTag("Key", "Value");
+                                }
+                            }
+                        });
+            }
+            JScrollPane alertTagsSp =
+                    createJScrollPane(Constant.messages.getString("alert.label.tags"));
+            alertTagsSp.setViewportView(alertTags);
+            alertTagsSp.addKeyListener(
+                    new KeyAdapter() {
```

Comment:
Can this use a lambda?

Reply:
Nope, KeyListener isn't a functional interface.